api/src/main/java/com/spotify/ffwd/output/BatchingPluginSink.java

```diff
@@ -70,6 +74,11 @@
     @Inject
     OutputPluginStatistics statistics;
 
+    @Inject
+    Supplier<BatchingStatistics> batchingStatisticsSupplier;
+
+    BatchingStatistics batchingStatistics = new NoopCoreStatistics().newBatching("");
```

Comment:
Could we instead provide this through the injected supplier? I.e. if `reportStatistics` is `false`, provide a `Supplier` that returns a noop impl?

Reply:
Thanks for the feedback! I've given it a try. Please see the separate commit :)